soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLAnalyzer.scala

```diff
@@ -581,11 +647,12 @@ class SoQLAnalyzer[RNS, CT, CV](typeInfo: TypeInfo2[CT, CV], functionInfo: Funct
 
                 val useQuery =
                   new Context(udfScope, Some(udfCanonicalName), Environment.empty, innerUdfParams)
-                    .analyzeStatement(None, parsed, None)
+                    .analyzeStatement(None, parsed, None, hiddenColumns)
 
                 FromStatement(
                   Select(
                     Distinctiveness.Indistinct,
+                    // note: we're adding the column mask here
```

Comment:
This is actually no longer true, going to remove the comment.  (We're adding the column mask - what I called hiddenColumns before realizing that was a terrible name and changing it - a few lines upward)